server/events/yaml/manifest_parser.go

```diff
@@ -0,0 +1,117 @@
+package yaml
+
+import (
+	"os"
+	"path/filepath"
+
+	version "github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events/yaml/raw"
+	"github.com/runatlantis/atlantis/server/events/yaml/valid"
+
+	"github.com/lyft/datum/pkg/manifest"
+)
+
+// ManifestYAMLFilename is the name of the config file for each repo.
+const ManifestYAMLFilename = "manifest.yaml"
+
+// ManifestParser find and converts all manifest files that contains
+// orchestration.roots
+type ManifestParser struct{}
+
+type manifestRoot struct {
+	dirPath  string
+	manifest *manifest.Manifest
+}
+
+// ToRepoCfg converts manifestRoots to valid.repoCfg configuration object.
+func (m *ManifestParser) ToRepoCfg(manifests []*manifestRoot) (valid.RepoCfg, error) {
+	var repoCfg valid.RepoCfg
+	repoCfg.Version = 3
+
+	for _, manifestRoot := range manifests {
+		roots := manifestRoot.manifest.Orchestration.Roots.List()
+		for _, root := range roots {
```

Comment:
Should we dedup based on root name/ root dir? Or do we do this in datum?

Reply:
No sure what you mean, can you expand on it?